src/patterns/gen-ai/aws-rag-appsync-stepfn-kendra/index.ts

```diff
@@ -0,0 +1,405 @@
+/**
+ *  Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License"). You may not use this file except in compliance
+ *  with the License. A copy of the License is located at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  or in the 'license' file accompanying this file. This file is distributed on an 'AS IS' BASIS, WITHOUT WARRANTIES
+ *  OR CONDITIONS OF ANY KIND, express or implied. See the License for the specific language governing permissions
+ *  and limitations under the License.
+ */
+import * as path from 'path';
+import * as cdk from 'aws-cdk-lib';
+import { Duration } from 'aws-cdk-lib';
+import * as appsync from 'aws-cdk-lib/aws-appsync';
+import * as cognito from 'aws-cdk-lib/aws-cognito';
+import { Table } from 'aws-cdk-lib/aws-dynamodb';
+import * as ec2 from 'aws-cdk-lib/aws-ec2';
+import { PolicyStatement } from 'aws-cdk-lib/aws-iam';
+import * as kendra from 'aws-cdk-lib/aws-kendra';
+import * as lambda from 'aws-cdk-lib/aws-lambda';
+import { DockerImageFunction } from 'aws-cdk-lib/aws-lambda';
+import * as logs from 'aws-cdk-lib/aws-logs';
+import { Bucket, BucketAccessControl, HttpMethods } from 'aws-cdk-lib/aws-s3';
+import { StateMachine } from 'aws-cdk-lib/aws-stepfunctions';
+import { Stack } from 'aws-cdk-lib/core';
+import { AwsCustomResource, AwsCustomResourcePolicy, PhysicalResourceId } from 'aws-cdk-lib/custom-resources';
+import { Construct } from 'constructs';
+import { KendraConstruct } from '../../../common/helpers/kendra-construct';
+import {
+  AddAwsServiceEndpoint,
+  createCheckJobsStatusFn,
+  createKendraStartDataSync,
+  createKendraWorkflowStepFunction,
+  createStepFunctionsExecutionHandlerRole,
+  createSyncRunTable, createUpdateKendraJobStatusFn, createDefaultIsolatedVpcProps, createS3FileUploader,
+} from '../../../common/helpers/kendra-helper';
+import { lambdaMemorySizeLimiter } from '../../../common/helpers/utils';
+import { buildVpc } from '../../../common/helpers/vpc-helper';
+
+/**
+ * The properties for the RagAppsyncStepfnKendraProps class.
+ */
+export interface RagAppsyncStepfnKendraProps {
```

Comment:
we can review the props, seems like the lambda ones are not used in the construct ?
also, seems like the construct is creating a new bucket, so we cannot provide an existing one ?